File: android/app/src/main/java/com/example/get_calley_test/models/Stats.java
package com.example.get_calley_test.models;

import java.util.List;

public class Stats {
    private int totalCalls;
    private int pendingCalls;
    private int completedCalls;
    private int scheduledCalls;
    private List<CallsByDay> callsByDay;

    public Stats() {}

    public int getTotalCalls() {
        return totalCalls;
    }

    public void setTotalCalls(int totalCalls) {
        this.totalCalls = totalCalls;
    }

    public int getPendingCalls() {
        return pendingCalls;
    }

    public void setPendingCalls(int pendingCalls) {
        this.pendingCalls = pendingCalls;
    }

    public int getCompletedCalls() {
        return completedCalls;
    }

    public void setCompletedCalls(int completedCalls) {
        this.completedCalls = completedCalls;
    }

    public int getScheduledCalls() {
        return scheduledCalls;
    }

    public void setScheduledCalls(int scheduledCalls) {
        this.scheduledCalls = scheduledCalls;
    }

    public List<CallsByDay> getCallsByDay() {
        return callsByDay;
    }

    public void setCallsByDay(List<CallsByDay> callsByDay) {
        this.callsByDay = callsByDay;
    }

    public static class CallsByDay {
        private String day;
        private int calls;

        public CallsByDay() {}

        public CallsByDay(String day, int calls) {
            this.day = day;
            this.calls = calls;
        }

        public String getDay() {
            return day;
        }

        public void setDay(String day) {
            this.day = day;
        }

        public int getCalls() {
            return calls;
        }

        public void setCalls(int calls) {
            this.calls = calls;
        }
    }
}

File: lib/models/stats.dart
class Stats {
  final int totalCalls;
  final int pendingCalls;
  final int completedCalls;
  final int scheduledCalls;
  final List<CallsByDay> callsByDay;

  Stats({
    required this.totalCalls,
    required this.pendingCalls,
    required this.completedCalls,
    required this.scheduledCalls,
    required this.callsByDay,
  });

  factory Stats.fromJson(Map<String, dynamic> json) {
    return Stats(
      totalCalls: json['totalCalls'] ?? 0,
      pendingCalls: json['pendingCalls'] ?? 0,
      completedCalls: json['completedCalls'] ?? 0,
      scheduledCalls: json['scheduledCalls'] ?? 0,
      callsByDay: (json['callsByDay'] as List<dynamic>?)
              ?.map((e) => CallsByDay.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'totalCalls': totalCalls,
      'pendingCalls': pendingCalls,
      'completedCalls': completedCalls,
      'scheduledCalls': scheduledCalls,
      'callsByDay': callsByDay.map((e) => e.toJson()).toList(),
    };
  }
}

class CallsByDay {
  final String day;
  final int calls;

  CallsByDay({
    required this.day,
    required this.calls,
  });

  factory CallsByDay.fromJson(Map<String, dynamic> json) {
    return CallsByDay(
      day: json['day'] ?? '',
      calls: json['calls'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'day': day,
      'calls': calls,
    };
  }
}
